recipes-nodes/rbuilder/rbuilder.inc

```diff
@@ -33,10 +34,11 @@ USERADD_PARAM:${PN} = "-r -s /bin/false -G eth rbuilder"
 do_install:append() {
     install -d ${D}${sysconfdir}/init.d
     install -m 0755 ${THISDIR}/init ${D}${sysconfdir}/init.d/rbuilder
-    
+
     # Install config files owned by root, readable by rbuilder
-    install -m 0744 ${THISDIR}/config ${D}${sysconfdir}/rbuilder.config
-    install -m 0744 ${THISDIR}/ofac.json ${D}${sysconfdir}/rbuilder.ofac.json
+    install -m 0644 ${THISDIR}/config.mustache ${D}${sysconfdir}/rbuilder.config.mustache
+    install -m 0640 ${THISDIR}/config.mustache ${D}${sysconfdir}/rbuilder.config
```

Comment:
This is a duplication and not needed
```suggestion
```

Reply:
It is not a duplicate. See my comment above. I pre-create the to-be rendered config to not set the correct permissions after it is rendered. It is set to 640 rather than 644 for the rest of the files as the config will contain secrets.

Reply:
wouldn't this pre-creation cause conflict with the render-config.sh? because it takes the mustache file and creates the real config. What would happen if there is already a file pre-created with the same name in that directory?

That's why I suggest to do the separation of config fetching and config rendering.
This way we can let the rbuilder renderer to render the config and then sets its permissions and ownership correctly.